### ISSUE TITLE
test: 通常推論とTTAの精度・速度を比較する

### ISSUE THREAD

**Issue by sharp231** [sharp231/classifier#7]
## 概要

同一の評価データを使用して、通常推論と現在のTTA推論の精度・速度を比較する。

TTAの継続、比率変更、廃止を実測結果に基づいて判断できるようにする。

## 背景

現在は次のcrop ratioで3回推論し、出力の平均値を最終結果として使用している。

```python
TTA_CROP_RATIOS = (1.0, 0.8, 0.65)
```

ただし、以下が確認できていない。

- TTAによってTop-1 Accuracyが改善するか
- Log Lossが改善するか
- クラスごとの精度に偏りがないか
- 右側を切り取る処理によって位置バイアスが発生しないか
- 推論時間の増加が許容できるか
- 現在のcrop ratioが妥当か

### Baseline

```python
crop_ratios = (1.0,)
```

### 現在のTTA

```python
crop_ratios = (1.0, 0.8, 0.65)
```

## 評価指標

最低限、次の値を記録する。

- 評価画像数
- Top-1 Accuracy
- Top-3 Accuracy
- Log Loss
- クラス別Accuracy
- 推論時間の平均値
- 推論時間の中央値
- 推論時間のp95
- 1画像あたりの推論回数
- モデルバージョン
- Python、TensorFlow、Kerasのバージョン
- 実行環境

可能であれば、TTAによって判定が改善した画像と悪化した画像の件数も記録する。

## 実装方針

- 推論とTTA平均処理を単体テスト可能な関数へ分離する
- 固定された評価データを使用する
- 評価データの出典とバージョンを記録する
- 評価処理を通常のWebテストから分離する
- 評価結果をJSONまたはMarkdownで保存する
- 大きな評価データ自体は無条件にGitへ追加しない

出力候補:

```text
reports/tta-evaluation.json
reports/tta-evaluation.md
```

## 判断基準

評価を実行する前に、次の判断基準を文書化する。

- どの程度の精度差を改善と判断するか
- 推論時間の増加をどこまで許容するか
- 特定クラスの大幅な悪化を許容するか
- 改善が確認できない場合にTTAを無効化するか
- crop ratioを変更して再評価するか

評価後に都合よく基準を変更しない。

## テスト対象

- 指定されたcrop ratioの回数だけモデルが呼び出される
- 3回の出力が要素単位で平均される
- 出力が信頼度順に並べ替えられる
- モデル出力の次元が不正な場合はエラーになる
- モデル出力数とラベル数が一致しない場合はエラーになる
- 空のcrop ratioを拒否する
- NaNまたは無限値を含む出力を安全に拒否する

## スコープ外

- モデルの再学習
- YOLOへの変更
- 新しいモデルの導入
- ユーザー画像の収集
- 本番環境でのA/Bテスト
- TTAが有効であることを前提としたUI実装

## 完了条件

- BaselineとTTAを同じ評価データで比較している
- 精度と速度の両方を記録している
- クラス別の悪化を確認できる
- 評価環境とモデルバージョンを記録している
- TTAを継続・変更・廃止する判断が文書化されている
- 推論ロジックの単体テストが成功する
- 全テストが成功する

## 依存関係

- 「本番とテストの推論前処理を共通化する」の完了が必要
